Realm/RLMSyncSubscription.mm

```diff
@@ -83,18 +200,89 @@ - (void)unsubscribe {
 }
 @end
 
-@implementation RLMResults (SyncSubscription)
+// RLMClassInfo stores pointers into the schema rather than owning the objects
+// it points to, so for a ClassInfo that's not part of the schema we need a
+// wrapper object that owns them
+RLMResultsSetInfo::RLMResultsSetInfo(__unsafe_unretained RLMRealm *const realm)
+: osObjectSchema(ObjectSchema(realm->_realm->read_group(), partial_sync::result_sets_type_name))
+, rlmObjectSchema([RLMObjectSchema objectSchemaForObjectStoreSchema:osObjectSchema])
+, info(realm, rlmObjectSchema, &osObjectSchema)
+{
+    rlmObjectSchema.accessorClass = [RLMSyncSubscriptionObject class];
+}
+
+RLMClassInfo& RLMResultsSetInfo::get(__unsafe_unretained RLMRealm *const realm) {
+    if (!realm->_resultsSetInfo) {
+        realm->_resultsSetInfo = std::make_unique<RLMResultsSetInfo>(realm);
+    }
+    return realm->_resultsSetInfo->info;
+}
+
+@interface RLMSubscriptionResults : RLMResults
+@end
 
+@implementation RLMSubscriptionResults
++ (instancetype)resultsWithRealm:(RLMRealm *)realm {
+    auto table = ObjectStore::table_for_object_type(realm->_realm->read_group(), partial_sync::result_sets_type_name);
+    if (!table) {
+        @throw RLMException(@"-[RLMRealm subscriptions] can only be called on a Realm using query-based sync");
+    }
+    // The server automatically adds a few subscriptions for the permissions
+    // types which we want to hide. They're just an implementation detail and
```

Comment:
In Java we actually just exposes them. Mostly because these methods might also be used for debugging and not showing all subscriptions can also make it hard to reason about the system, especially if the API's are used by us internally. I do agree that if a user by accident deletes them they will probably be very surprised, but IMO the chance is pretty slim and will probably be detected pretty quick.